dialogue-clients/src/test/java/com/palantir/dialogue/otherpackage/DialogueClientsIntegrationTest.java

```diff
@@ -341,6 +341,21 @@ void test_unlimited_timeouts() {
                 .doesNotThrowAnyException();
     }
 
+    @Test
+    void test_mesh_uri() {
+        undertowHandler = exchange -> exchange.setStatusCode(204);
+        SampleServiceBlocking client = DialogueClients.create(Refreshable.only(null))
+                .withUserAgent(TestConfigurations.AGENT)
+                .withMaxNumRetries(0)
+                .getNonReloading(
+                        SampleServiceBlocking.class,
+                        ServiceConfiguration.builder()
+                                .addUris("mesh-" + getUri(undertow))
+                                .security(TestConfigurations.SSL_CONFIG)
+                                .build());
+        assertThatCode(client::voidToVoid).as("request should succeed").doesNotThrowAnyException();
+    }
```

Comment:
Note that this passes on develop, but I figured we should have full test coverage of this case just in case!